cdmtaskservice/dockerimages.py

```diff
@@ -2,29 +2,151 @@
 Contains code for querying container information without access to a docker server
 """
 
+# Trying to find a simple way to handle this was surprisingly frustrating. I spent a half
+# day or so looking at alternatives and the crane CLI seems like the least bad option. I tried
+# a number of things, most of which I don't remember, including http requests (had to be tailored
+# to each repository somewhat) and https://github.com/davedoesdev/dxf (couldn't get anonymous
+# access to work). If there's a python library that can take crane's place that'd be great but
```

Comment:
can we use docker client? It will pull image but we don't need to install `crane`, validate image name/hostname, etc. 
  
```
In [5]: import docker
   ...: 
   ...: client = docker.from_env()

In [6]: image = client.images.pull("kbase/workspace_del
   ...: uxe:0.10.2-hotfix")

In [7]: image.attrs['RepoDigests'][0].split('@')[1]
Out[7]: 'sha256:285b1229730192eac5b502c44bfffff98e9840057821d1a26bed47a05bc44874'
```

Reply:
That's the first thing I tried but it won't work without a docker service, and the cdm task service won't have access to one.

Reply:
👍